### DISCUSSION TITLE
Artisan v2.8.5 Release Candidate

### DISCUSSION THREAD

**Discussion by MAKOMO** [artisan-roaster-scope/artisan#1312] [Announcement]
Dear all,

we would love if you could give the Release Candidate for our next Artisan version a quick test and report back if this one works for you. We still can fix issues before the release, planned for the end of this month.

[https://github.com/artisan-roaster-scope/artisan/releases/tag/v2.8.5-RC](https://github.com/artisan-roaster-scope/artisan/releases/tag/v2.8.5-RC)

What changed:

* extra devices added to Roast Comparator
* equal and not equal temperature conditions added to alarm rules
* new font graph font options Comic Neue (a redesigned Comic Sans) and xkcd Script
* additional machine support for Bühler Roastmaster, Joper, Cogen, Typhoon, Carmomaq
* support Phidget DAQ1000
* fat macOS builds running native on Apple Silicon (M1, M2, ..) incl. macOS 14 Sonoma
* adds back experimental Raspbian OS build (bookworm 64bit)
* many bug fixes as well as performance and stability improvements

Find the complete list of additions, changes and fixes in our [Release History](https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/ReleaseHistory.md)

Awaiting your reports!

**Comment by mo3adz** [comment]
Greetings

I've tried it on my Joper BPR and it includes PLC. I connect it directly to the Ethernet cable and choose Joper PLC from the Config > machine. 
This message appears ( modbus error : failed to connect ) 

![image](https://github.com/artisan-roaster-scope/artisan/assets/151010166/392a77df-5971-49a5-a6ba-43d620a99d6c)

**Reply by MAKOMO** [reply]
Did you contact Joper to ask if your machine is supported? Did you choose the correct IP address in Artisan? Is that IP address accessible from your computer running Artisan?

**Reply by mo3adz** [reply]
I already contacted Joper and my machine is supported. And I've chosen the right IP address

**Reply by poundy** [reply]
but is your PC's IP settings and the IP address of the roaster able to communicate? If you're not sure what that means, tell us the two IP addresses in question, or refer to the guidance on https://artisan-scope.org/machines/toper/ that talks about the requirement

**Reply by mo3adz** [reply]
It's Joper not toper and I already made sure I'm using the correct IP's

**Reply by MAKOMO** [reply]
Well, it is obviously a network issue. Artisan is not able to connect to your machine at all as if the wire is disconnected. Nothing that can be done on the Artisan side about this. Sorry.

**Reply by mo3adz** [reply]
I will change the cable and we will hope the best

**Reply by MAKOMO** [reply]
I don't believe it is a cable issue, but rather a network configuration issue on your DHCP server or your PC. Ask a network expert for support (or Joper?).

**Reply by poundy** [reply]
the reason I pointed to the Toper page is that it discusses the networking aspect. I too, like Makomo, do not think it's a cable issue but your configuration. That's why I asked you to provide the IP addresses so we could help explain it to you

**Comment by jwarthman** [comment]
Makomo,
Thank you! With this 2.8.5 (0) RC I'm now able to use my MacBook Pro M2 running macOS Sonoma 14.1.1 and a Hottop 2K+.

I've noted two anomalies. On the first roast with this version, I had made some adjustments, such as Roast Properties > "Open on Drop" and others. What happened was, when I dropped the beans the door opened and some beans went into the cooling tray. But moments later the drum and stirrer stopped turning! I fumbled long enough for the beans that stayed in the drum to go into second crack and they were smoking like crazy! When I finally used the Hottop's emergency drop lever the drum started turning and the remaining beans dropped - although way over-cooked. I disabled "Open on Drop" and the problem did not recur on subsequent roasts.

The second issue is more persistent. On every roast, I need to unplug then re-plug the USB-C cable for Artisan to see the Hottop. The sequence of events is:

* Launch Artisan by opening an .alog file for the upcoming roast
* Upper left corner says "Hottop disconnected".
* Unplug USB-C, then re-plug and this changes to "Hottop connected".

Please let me know if you need additional information.

Best Regards,

Jim Warthman

**Comment by Terracotta-6** [comment]
extra device curves to Comparator , Chinese character coding conversion problem.
![2f3a37283743488b1a48543443faac4](https://github.com/artisan-roaster-scope/artisan/assets/149906183/30d993ff-dd68-4918-8580-7dec1052d8c2)

**Reply by MAKOMO** [reply]
Good catch! Fixed (Commit 02d78142d7d3dae9708ecc829795e10bc2c1cbe7)

**Reply by Terracotta-6** [reply]
Incomplete conversion.
![7654e3c3cc1bbcc82748d7dc84bd8a4](https://github.com/artisan-roaster-scope/artisan/assets/149906183/3177b5cd-19d8-46d7-833b-2207e0f12505)

**Reply by MAKOMO** [reply]
Another good one. Should be fixed by Commit 7b034bdab9a5643450354ba1c323dcb588dcd8ac

**Comment by MAKOMO** [comment]
> Makomo, Thank you! With this 2.8.5 (0) RC I'm now able to use my MacBook Pro M2 running macOS Sonoma 14.1.1 and a Hottop 2K+.
> 
> I've noted two anomalies. On the first roast with this version, I had made some adjustments, such as Roast Properties > "Open on Drop" and others. What happened was, when I dropped the beans the door opened and some beans went into the cooling tray. But moments later the drum and stirrer stopped turning! 

Once you click OFF or the communication between the machine and Artisan gets disconnected, the machine turns off. This is done by the Hottop firmware. Nothing we can do about.

>  I disabled "Open on Drop" and the problem did not recur on subsequent roasts.

This is unlikely to be related as it has nothing to do with communication and happens in a separate thread.

Most likely you experience random communication issues. We found the Hottop to be very sensitive to Electromagnetic interference (EMI). See [Artisan spontaneously disconnects from HotTop KN-8828B-2K+ with MacBook Air M2](https://github.com/artisan-roaster-scope/artisan/issues/1257) for a related discussion and some counter measures.

> The second issue is more persistent. On every roast, I need to unplug then re-plug the USB-C cable for Artisan to see the Hottop. The sequence of events is:
> 
> * Launch Artisan by opening an .alog file for the upcoming roast
> * Upper left corner says "Hottop disconnected".
> * Unplug USB-C, then re-plug and this changes to "Hottop connected".

I cannot duplicate this using my machine. This seems must be an OS issue. Artisan connects to the serial port as specified. If this serial port object is provided by the OS on connecting the Hottop and should not disappear as long as the Hottop remains connected. However, this issue might be connected also to some EMI issues. If the firmware of the Hottop crashes the OS will most likely remove the broken serial port and the Hottop seems not to re-establish it on restart (if it does restart at all), but only on re-connecting the USB wire.

Nothing that can be done on any of this on the Artisan side. Sorry.

**Reply by jwarthman** [reply]
Makomo,
Thanks for your quick reply!

>  Once you click OFF or the communication between the machine and Artisan gets disconnected, the machine turns off. This is done by the Hottop firmware. Nothing we can do about.

Over 6 roasts this only happened once. I'll keep an eye on this and if it happens again I'll try to capture more detail. Would it be helpful to enable device logging, and just leave it on?

> I cannot duplicate this using my machine. This seems must be an OS issue. Artisan connects to the serial port as specified. If this serial port object is provided by the OS on connecting the Hottop and should not disappear as long as the Hottop remains connected. However, this issue might be connected also to some EMI issues. If the firmware of the Hottop crashes the OS will most likely remove the broken serial port and the Hottop seems not to re-establish it on restart (if it does restart at all), but only on re-connecting the USB wire.

Right, I never had this issue on the prior OS version. I don't think EMI is involved because the Hottop and computer are in the same locations as previously, and not near any EMI sources - unless Sonoma cause the Mac to emit RF energy differently than Veentura.

Would it be helpful to enable device logging to shed more light on the communication failure?

Thanks,

Jim

**Reply by MAKOMO** [reply]
If it is not EMI it could be also a power issue. Opening the drop gate and at the same time starting the stirrer can bring down the voltage on this little Hottop electronic to give up. 

A device log is most likely not productive as it does not show the machines side of things as this simple protocol does not contain any confirmation messages. You could attach your Artisan log file here for me take a quick look, thus I also do not expect to seem much in that one.

**Comment by zihyer** [comment]
I'll be able to test the DAQ1301 functionality when my parts come in.  Will report back soon.  Thanks for all your hard work!
